enricher/api/src/main/java/io/fabric8/maven/enricher/api/EnricherContext.java

```diff
@@ -1,175 +1,167 @@
+     */
+    String getDependencyVersion(String groupId, String artifactId);
 
-    public OpenShiftDependencyResources getOpenshiftDependencyResources() {
-        return openshiftDependencyResources;
-    }
+    /**
+     * Gets Compile Classloader.
+     * @return Classloader.
+     */
+    ClassLoaderWrapper getCompileClassLoader();
 
     /**
-     * Returns true if we are in watch mode
+     * Gets Test ClassLoader
+     * @return ClassLoader
      */
-    public boolean isWatchMode() {
-        try {
-            return runningWithGoal("fabric8:watch-spring-boot", "fabric8:watch");
-        } catch (MojoExecutionException e) {
-            throw new IllegalStateException("Cannot determine maven goals", e);
-        }
-    }
+    ClassLoaderWrapper getTestClassLoader();
 
     /**
-     * Returns true if maven is running with any of the given goals
+     * Gets Scm information.
+     * @return scm info.
      */
-    public boolean runningWithGoal(String... goals) throws MojoExecutionException {
-        for (String goal : goals) {
-            if (goalFinder.runningWithGoal(project, session,  goal)) {
-                return true;
-            }
-        }
-        return false;
-    }
-
-    // =======================================================================================================
-    public static class Builder {
-
-        private EnricherContext ctx = new EnricherContext();
-
-        public Builder session(MavenSession session) {
-            ctx.session = session;
-            return this;
-        };
-
-        public Builder goalFinder(GoalFinder goalFinder) {
-            ctx.goalFinder = goalFinder;
-            return this;
-        }
-
-        public Builder log(Logger log) {
-            ctx.log = log;
-            return this;
-        }
-
-        public Builder project(MavenProject project) {
-            ctx.project = project;
-            return this;
-        }
-
-        public Builder config(ProcessorConfig config) {
-            ctx.config = config;
-            return this;
-        }
-
-        public Builder resources(ResourceConfig resources) {
-            ctx.resources = resources;
-            return this;
-        }
-
-        public Builder images(List<ImageConfiguration> images) {
-            ctx.images = images;
-            return this;
-        }
-
-        public Builder namespace(String namespace) {
-            ctx.namespace = namespace;
-            return this;
-        }
-
-        public Builder useProjectClasspath(boolean useProjectClasspath) {
-            ctx.useProjectClasspath = useProjectClasspath;
-            return this;
-        }
-
-        public Builder openshiftDependencyResources(OpenShiftDependencyResources openShiftDependencyResources) {
-            ctx.openshiftDependencyResources = openShiftDependencyResources;
-            return this;
-        }
-
-        public EnricherContext build() {
-            return ctx;
-        }
-
-    }
+    Scm getScm();
```

Comment:
This could be either reused in the ScmEnricher (currently only used in the IconEnricher), or removed so that the IconEnricher should also probe for a MavenEnricherContext.

Not 100% sure, but I really would like to keep the context as slim as possible and if something is only used by one single enricher, I tend to favor the cast.